### PR TITLE
Updates for v2

### PR DESCRIPTION
**Pull request by dfcreative**
Hi @bpostlethwaite 

The non-breaking update includes:

* [x] reduced lib size #13 with removed deps
* [x] fixed duplicated colors, maps are smoother now #9 
* [x] simplified tests/example
* [x] extra formats
* [x] fix tests

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -66,59 +67,72 @@ function createColormap (spec) {
...
-    /*
-     * Add alpha channel to the map
-     */
-    if (alpha[0] < 0) alpha[0] = 0;
-    if (alpha[1] < 0) alpha[0] = 0;
-    if (alpha[0] > 1) alpha[0] = 1;
-    if (alpha[1] > 1) alpha[0] = 1;
+    // Add alpha channel to the map
+    alpha[0] = clamp(alpha[0], 0, 1);
+    alpha[1] = clamp(alpha[1], 0, 1);
```
Comment (bpostlethwaite) [inline index.js]: using clamp saves 2 lines of code _at this location_ but adds a `require` line and a line to package json. Total lines saved === 0.
However the complexity of the code has increased. Not a major issue but I vote to to keep the manual clamping in place.

File: test.js [hunk 2]
```diff
@@ -63,3 +82,7 @@ test('alphamap values are computed independently between runs', function(t) {
 
     t.end();
 });
+
+test('repeating values', function(t) {
+    t.end()
```
Comment (bpostlethwaite) [inline test.js]: ?

File: index.js [hunk 3]
```diff
@@ -23,9 +24,9 @@ function createColormap (spec) {
         b = [],
         a = [];
 
-    if ( !at.isPlainObject(spec) ) spec = {};
+    if ( !isPlainObject(spec) ) spec = {};
```
Comment (bpostlethwaite) [inline index.js]: We could cut this dep as well. For example if a user wants to pass in an instance of a prototype as a specifcation object (TypeScript compat for example) they should be able to. If it walks like a duck...

A check for falsey values is probably sufficient.

**Comment by bpostlethwaite** [comment]
Great! I think we can remove 2 dependencies and actually improve readability and the API in the process.

**Comment by dfcreative** [comment]
@bpostlethwaite done!

**Comment by dfcreative** [comment]
@bpostlethwaite I would make this update a minor release since API is not changed and make API change in major release (not ready yet).

**Pull request merged**